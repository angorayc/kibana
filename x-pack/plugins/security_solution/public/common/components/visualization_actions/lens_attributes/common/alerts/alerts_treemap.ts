/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
import type { GetLensAttributes } from '../../../types';
import { buildAlertsOptionsFilters } from './utils';

export const getAlertsTreemapLensAttributes: GetLensAttributes = (
  stackByField = 'kibana.alert.rule.name',
  extraOptions = {
    showOnlyThreatIndicatorAlerts: false,
    showBuildingBlockAlerts: false,
  }
) => ({
  title: 'Alerts',
  description: '',
  visualizationType: 'lnsPie',
  state: {
    visualization: {
      shape: 'treemap',
      layers: [
        {
          layerId: '4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b',
          primaryGroups: [
            '2881fedd-54b7-42ba-8c97-5175dec86166',
            '75ce269b-ee9c-4c7d-a14e-9226ba0fe059',
          ],
          metrics: ['f04a71a3-399f-4d32-9efc-8a005e989991'],
          numberDisplay: 'value',
          categoryDisplay: 'default',
          legendDisplay: 'show',
          nestedLegend: true,
          layerType: 'data',
          legendSize: 'xlarge',
          legendPosition: 'left',
        },
      ],
    },
    query: {
      query: '',
      language: 'kuery',
    },
    filters: buildAlertsOptionsFilters(extraOptions),
    datasourceStates: {
      formBased: {
        layers: {
          '4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b': {
            columns: {
              '2881fedd-54b7-42ba-8c97-5175dec86166': {
                label: `Top values of ${stackByField}`,
                dataType: 'string',
                operationType: 'terms',
                scale: 'ordinal',
                sourceField: stackByField,
                isBucketed: true,
                params: {
                  size: 1000,
                  orderBy: {
                    type: 'column',
                    columnId: 'f04a71a3-399f-4d32-9efc-8a005e989991',
                  },
                  orderDirection: 'desc',
                  otherBucket: true,
                  missingBucket: false,
                  parentFormat: {
                    id: 'terms',
                  },
                  include: [],
                  exclude: [],
                  includeIsRegex: false,
                  excludeIsRegex: false,
                },
              },
              '75ce269b-ee9c-4c7d-a14e-9226ba0fe059': {
                label: `Top values of ${extraOptions.breakdownField}`,
                dataType: 'string',
                operationType: 'terms',
                scale: 'ordinal',
                sourceField: extraOptions.breakdownField,
                isBucketed: true,
                params: {
                  size: 1000,
                  orderBy: {
                    type: 'column',
                    columnId: 'f04a71a3-399f-4d32-9efc-8a005e989991',
                  },
                  orderDirection: 'desc',
                  otherBucket: true,
                  missingBucket: false,
                  parentFormat: {
                    id: 'terms',
                  },
                  include: [],
                  exclude: [],
                  includeIsRegex: false,
                  excludeIsRegex: false,
                },
              },
              'f04a71a3-399f-4d32-9efc-8a005e989991': {
                label: 'Maximum of kibana.alert.risk_score',
                dataType: 'number',
                operationType: 'max',
                sourceField: 'kibana.alert.risk_score',
                isBucketed: false,
                scale: 'ratio',
                params: {
                  emptyAsNull: true,
                },
              },
            },
            columnOrder: [
              '2881fedd-54b7-42ba-8c97-5175dec86166',
              '75ce269b-ee9c-4c7d-a14e-9226ba0fe059',
              'f04a71a3-399f-4d32-9efc-8a005e989991',
            ],
            sampling: 1,
            incompleteColumns: {},
          },
        },
      },
      textBased: {
        layers: {},
      },
    },
    internalReferences: [],
    adHocDataViews: {},
  },
  references: [
    {
      type: 'index-pattern',
      id: '{dataViewId}',
      name: 'indexpattern-datasource-layer-4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b',
    },
  ],
});
