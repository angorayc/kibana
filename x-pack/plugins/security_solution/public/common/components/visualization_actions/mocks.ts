/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
import type { LensAttributes } from './types';

export const mockAttributes: LensAttributes = {
  description: '',
  state: {
    datasourceStates: {
      formBased: {
        layers: {
          '416b6fad-1923-4f6a-a2df-b223bb287e30': {
            columnOrder: ['b00c65ea-32be-4163-bfc8-f795b1ef9d06'],
            columns: {
              'b00c65ea-32be-4163-bfc8-f795b1ef9d06': {
                customLabel: true,
                dataType: 'number',
                isBucketed: false,
                label: ' ',
                operationType: 'unique_count',
                scale: 'ratio',
                sourceField: 'host.name',
              },
            },
            incompleteColumns: {},
          },
        },
      },
    },
    filters: [
      {
        meta: {
          type: 'phrases',
          key: '_index',
          params: ['packetbeat-*'],
          alias: null,
          negate: false,
          disabled: false,
        },
        query: {
          bool: {
            should: [{ match_phrase: { _index: 'packetbeat-*' } }],
            minimum_should_match: 1,
          },
        },
      },
    ],
    query: { query: '', language: 'kuery' },
    visualization: {
      accessor: 'b00c65ea-32be-4163-bfc8-f795b1ef9d06',
      layerId: '416b6fad-1923-4f6a-a2df-b223bb287e30',
      layerType: 'data',
    },
  },
  title: '',
  visualizationType: 'lnsLegacyMetric',
  references: [
    {
      id: 'security-solution-default',
      name: 'indexpattern-datasource-current-indexpattern',
      type: 'index-pattern',
    },
    {
      id: 'security-solution-default',
      name: 'indexpattern-datasource-layer-416b6fad-1923-4f6a-a2df-b223bb287e30',
      type: 'index-pattern',
    },
  ],
};
