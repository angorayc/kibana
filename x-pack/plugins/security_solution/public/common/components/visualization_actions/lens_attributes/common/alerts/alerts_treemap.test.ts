/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
import { renderHook } from '@testing-library/react-hooks';
import { wrapper } from '../../../mocks';

import { useLensAttributes } from '../../../use_lens_attributes';

import { getAlertsTreemapLensAttributes } from './alerts_treemap';

jest.mock('uuid', () => ({
  v4: jest.fn().mockReturnValue('4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b'),
}));

jest.mock('../../../../../containers/sourcerer', () => ({
  useSourcererDataView: jest.fn().mockReturnValue({
    dataViewId: 'security-solution-my-test',
    indicesExist: true,
    selectedPatterns: ['signal-index'],
  }),
}));

jest.mock('../../../../../utils/route/use_route_spy', () => ({
  useRouteSpy: jest.fn().mockReturnValue([
    {
      pageName: 'alerts',
    },
  ]),
}));

describe('getAlertsTreemapLensAttributes', () => {
  it('should render without extra options', () => {
    const { result } = renderHook(
      () =>
        useLensAttributes({
          getLensAttributes: getAlertsTreemapLensAttributes,
          stackByField: 'event.category',
        }),
      { wrapper }
    );

    expect(result?.current).toMatchSnapshot();
  });

  it('should render with extra options - filters', () => {
    const { result } = renderHook(
      () =>
        useLensAttributes({
          extraOptions: {
            filters: [
              {
                meta: {
                  type: 'phrases',
                  key: '_index',
                  params: ['.alerts-security.alerts-default'],
                  alias: null,
                  negate: false,
                  disabled: false,
                },
                query: {
                  bool: {
                    should: [
                      {
                        match_phrase: {
                          _index: '.alerts-security.alerts-default',
                        },
                      },
                    ],
                    minimum_should_match: 1,
                  },
                },
              },
            ],
          },
          getLensAttributes: getAlertsTreemapLensAttributes,
          stackByField: 'event.category',
        }),
      { wrapper }
    );

    expect(result?.current).toMatchSnapshot();
  });

  it('should render with extra options - breakdownField', () => {
    const { result } = renderHook(
      () =>
        useLensAttributes({
          extraOptions: { breakdownField: 'agent.type' },
          getLensAttributes: getAlertsTreemapLensAttributes,
          stackByField: 'event.category',
        }),
      { wrapper }
    );

    expect(result?.current).toMatchSnapshot();
  });
});
