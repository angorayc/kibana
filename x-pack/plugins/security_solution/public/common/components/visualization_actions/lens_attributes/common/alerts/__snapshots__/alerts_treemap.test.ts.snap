// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAlertsTreemapLensAttributes should render with extra options - breakdownField 1`] = `
Object {
  "description": "",
  "references": Array [
    Object {
      "id": "security-solution-my-test",
      "name": "indexpattern-datasource-layer-4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
      "type": "index-pattern",
    },
  ],
  "state": Object {
    "adHocDataViews": Object {},
    "datasourceStates": Object {
      "formBased": Object {
        "layers": Object {
          "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b": Object {
            "columnOrder": Array [
              "2881fedd-54b7-42ba-8c97-5175dec86166",
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
              "f04a71a3-399f-4d32-9efc-8a005e989991",
            ],
            "columns": Object {
              "2881fedd-54b7-42ba-8c97-5175dec86166": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of event.category",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": "event.category",
              },
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of agent.type",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": "agent.type",
              },
              "f04a71a3-399f-4d32-9efc-8a005e989991": Object {
                "dataType": "number",
                "isBucketed": false,
                "label": "Maximum of kibana.alert.risk_score",
                "operationType": "max",
                "params": Object {
                  "emptyAsNull": true,
                },
                "scale": "ratio",
                "sourceField": "kibana.alert.risk_score",
              },
            },
            "incompleteColumns": Object {},
            "sampling": 1,
          },
        },
      },
      "textBased": Object {
        "layers": Object {},
      },
    },
    "filters": Array [
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "index": ".alerts-security.alerts-id",
          "key": "kibana.alert.building_block_type",
          "negate": true,
          "type": "exists",
        },
        "query": Object {
          "exists": Object {
            "field": "kibana.alert.building_block_type",
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "host.id",
          "negate": false,
          "params": Object {
            "query": "123",
          },
          "type": "phrase",
        },
        "query": Object {
          "match_phrase": Object {
            "host.id": "123",
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "_index",
          "negate": false,
          "params": Array [
            "signal-index",
          ],
          "type": "phrases",
        },
        "query": Object {
          "bool": Object {
            "minimum_should_match": 1,
            "should": Array [
              Object {
                "match_phrase": Object {
                  "_index": "signal-index",
                },
              },
            ],
          },
        },
      },
    ],
    "internalReferences": Array [],
    "query": Object {
      "language": "kql",
      "query": "host.name: *",
    },
    "visualization": Object {
      "layers": Array [
        Object {
          "categoryDisplay": "default",
          "layerId": "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
          "layerType": "data",
          "legendDisplay": "show",
          "legendPosition": "left",
          "legendSize": "xlarge",
          "metrics": Array [
            "f04a71a3-399f-4d32-9efc-8a005e989991",
          ],
          "nestedLegend": true,
          "numberDisplay": "value",
          "primaryGroups": Array [
            "2881fedd-54b7-42ba-8c97-5175dec86166",
            "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
          ],
        },
      ],
      "shape": "treemap",
    },
  },
  "title": "Alerts",
  "visualizationType": "lnsPie",
}
`;

exports[`getAlertsTreemapLensAttributes should render with extra options - showBuildingBlockAlerts 1`] = `
Object {
  "description": "",
  "references": Array [
    Object {
      "id": "security-solution-my-test",
      "name": "indexpattern-datasource-layer-4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
      "type": "index-pattern",
    },
  ],
  "state": Object {
    "adHocDataViews": Object {},
    "datasourceStates": Object {
      "formBased": Object {
        "layers": Object {
          "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b": Object {
            "columnOrder": Array [
              "2881fedd-54b7-42ba-8c97-5175dec86166",
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
              "f04a71a3-399f-4d32-9efc-8a005e989991",
            ],
            "columns": Object {
              "2881fedd-54b7-42ba-8c97-5175dec86166": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of event.category",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": "event.category",
              },
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of undefined",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": undefined,
              },
              "f04a71a3-399f-4d32-9efc-8a005e989991": Object {
                "dataType": "number",
                "isBucketed": false,
                "label": "Maximum of kibana.alert.risk_score",
                "operationType": "max",
                "params": Object {
                  "emptyAsNull": true,
                },
                "scale": "ratio",
                "sourceField": "kibana.alert.risk_score",
              },
            },
            "incompleteColumns": Object {},
            "sampling": 1,
          },
        },
      },
      "textBased": Object {
        "layers": Object {},
      },
    },
    "filters": Array [
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "host.id",
          "negate": false,
          "params": Object {
            "query": "123",
          },
          "type": "phrase",
        },
        "query": Object {
          "match_phrase": Object {
            "host.id": "123",
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "_index",
          "negate": false,
          "params": Array [
            "signal-index",
          ],
          "type": "phrases",
        },
        "query": Object {
          "bool": Object {
            "minimum_should_match": 1,
            "should": Array [
              Object {
                "match_phrase": Object {
                  "_index": "signal-index",
                },
              },
            ],
          },
        },
      },
    ],
    "internalReferences": Array [],
    "query": Object {
      "language": "kql",
      "query": "host.name: *",
    },
    "visualization": Object {
      "layers": Array [
        Object {
          "categoryDisplay": "default",
          "layerId": "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
          "layerType": "data",
          "legendDisplay": "show",
          "legendPosition": "left",
          "legendSize": "xlarge",
          "metrics": Array [
            "f04a71a3-399f-4d32-9efc-8a005e989991",
          ],
          "nestedLegend": true,
          "numberDisplay": "value",
          "primaryGroups": Array [
            "2881fedd-54b7-42ba-8c97-5175dec86166",
            "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
          ],
        },
      ],
      "shape": "treemap",
    },
  },
  "title": "Alerts",
  "visualizationType": "lnsPie",
}
`;

exports[`getAlertsTreemapLensAttributes should render with extra options - showOnlyThreatIndicatorAlerts 1`] = `
Object {
  "description": "",
  "references": Array [
    Object {
      "id": "security-solution-my-test",
      "name": "indexpattern-datasource-layer-4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
      "type": "index-pattern",
    },
  ],
  "state": Object {
    "adHocDataViews": Object {},
    "datasourceStates": Object {
      "formBased": Object {
        "layers": Object {
          "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b": Object {
            "columnOrder": Array [
              "2881fedd-54b7-42ba-8c97-5175dec86166",
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
              "f04a71a3-399f-4d32-9efc-8a005e989991",
            ],
            "columns": Object {
              "2881fedd-54b7-42ba-8c97-5175dec86166": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of event.category",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": "event.category",
              },
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of undefined",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": undefined,
              },
              "f04a71a3-399f-4d32-9efc-8a005e989991": Object {
                "dataType": "number",
                "isBucketed": false,
                "label": "Maximum of kibana.alert.risk_score",
                "operationType": "max",
                "params": Object {
                  "emptyAsNull": true,
                },
                "scale": "ratio",
                "sourceField": "kibana.alert.risk_score",
              },
            },
            "incompleteColumns": Object {},
            "sampling": 1,
          },
        },
      },
      "textBased": Object {
        "layers": Object {},
      },
    },
    "filters": Array [
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "kibana.alert.rule.type",
          "negate": false,
          "params": Object {
            "query": "threat_match",
          },
          "type": "phrase",
        },
        "query": Object {
          "match_phrase": Object {
            "kibana.alert.rule.type": "threat_match",
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "index": ".alerts-security.alerts-id",
          "key": "kibana.alert.building_block_type",
          "negate": true,
          "type": "exists",
        },
        "query": Object {
          "exists": Object {
            "field": "kibana.alert.building_block_type",
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "host.id",
          "negate": false,
          "params": Object {
            "query": "123",
          },
          "type": "phrase",
        },
        "query": Object {
          "match_phrase": Object {
            "host.id": "123",
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "_index",
          "negate": false,
          "params": Array [
            "signal-index",
          ],
          "type": "phrases",
        },
        "query": Object {
          "bool": Object {
            "minimum_should_match": 1,
            "should": Array [
              Object {
                "match_phrase": Object {
                  "_index": "signal-index",
                },
              },
            ],
          },
        },
      },
    ],
    "internalReferences": Array [],
    "query": Object {
      "language": "kql",
      "query": "host.name: *",
    },
    "visualization": Object {
      "layers": Array [
        Object {
          "categoryDisplay": "default",
          "layerId": "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
          "layerType": "data",
          "legendDisplay": "show",
          "legendPosition": "left",
          "legendSize": "xlarge",
          "metrics": Array [
            "f04a71a3-399f-4d32-9efc-8a005e989991",
          ],
          "nestedLegend": true,
          "numberDisplay": "value",
          "primaryGroups": Array [
            "2881fedd-54b7-42ba-8c97-5175dec86166",
            "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
          ],
        },
      ],
      "shape": "treemap",
    },
  },
  "title": "Alerts",
  "visualizationType": "lnsPie",
}
`;

exports[`getAlertsTreemapLensAttributes should render with extra options - status 1`] = `
Object {
  "description": "",
  "references": Array [
    Object {
      "id": "security-solution-my-test",
      "name": "indexpattern-datasource-layer-4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
      "type": "index-pattern",
    },
  ],
  "state": Object {
    "adHocDataViews": Object {},
    "datasourceStates": Object {
      "formBased": Object {
        "layers": Object {
          "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b": Object {
            "columnOrder": Array [
              "2881fedd-54b7-42ba-8c97-5175dec86166",
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
              "f04a71a3-399f-4d32-9efc-8a005e989991",
            ],
            "columns": Object {
              "2881fedd-54b7-42ba-8c97-5175dec86166": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of event.category",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": "event.category",
              },
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of undefined",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": undefined,
              },
              "f04a71a3-399f-4d32-9efc-8a005e989991": Object {
                "dataType": "number",
                "isBucketed": false,
                "label": "Maximum of kibana.alert.risk_score",
                "operationType": "max",
                "params": Object {
                  "emptyAsNull": true,
                },
                "scale": "ratio",
                "sourceField": "kibana.alert.risk_score",
              },
            },
            "incompleteColumns": Object {},
            "sampling": 1,
          },
        },
      },
      "textBased": Object {
        "layers": Object {},
      },
    },
    "filters": Array [
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "kibana.alert.workflow_status",
          "negate": false,
          "params": Object {
            "query": "open",
          },
          "type": "phrase",
        },
        "query": Object {
          "match_phrase": Object {
            "kibana.alert.workflow_status": "open",
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "index": ".alerts-security.alerts-id",
          "key": "kibana.alert.building_block_type",
          "negate": true,
          "type": "exists",
        },
        "query": Object {
          "exists": Object {
            "field": "kibana.alert.building_block_type",
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "host.id",
          "negate": false,
          "params": Object {
            "query": "123",
          },
          "type": "phrase",
        },
        "query": Object {
          "match_phrase": Object {
            "host.id": "123",
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "_index",
          "negate": false,
          "params": Array [
            "signal-index",
          ],
          "type": "phrases",
        },
        "query": Object {
          "bool": Object {
            "minimum_should_match": 1,
            "should": Array [
              Object {
                "match_phrase": Object {
                  "_index": "signal-index",
                },
              },
            ],
          },
        },
      },
    ],
    "internalReferences": Array [],
    "query": Object {
      "language": "kql",
      "query": "host.name: *",
    },
    "visualization": Object {
      "layers": Array [
        Object {
          "categoryDisplay": "default",
          "layerId": "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
          "layerType": "data",
          "legendDisplay": "show",
          "legendPosition": "left",
          "legendSize": "xlarge",
          "metrics": Array [
            "f04a71a3-399f-4d32-9efc-8a005e989991",
          ],
          "nestedLegend": true,
          "numberDisplay": "value",
          "primaryGroups": Array [
            "2881fedd-54b7-42ba-8c97-5175dec86166",
            "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
          ],
        },
      ],
      "shape": "treemap",
    },
  },
  "title": "Alerts",
  "visualizationType": "lnsPie",
}
`;

exports[`getAlertsTreemapLensAttributes should render without extra options 1`] = `
Object {
  "description": "",
  "references": Array [
    Object {
      "id": "security-solution-my-test",
      "name": "indexpattern-datasource-layer-4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
      "type": "index-pattern",
    },
  ],
  "state": Object {
    "adHocDataViews": Object {},
    "datasourceStates": Object {
      "formBased": Object {
        "layers": Object {
          "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b": Object {
            "columnOrder": Array [
              "2881fedd-54b7-42ba-8c97-5175dec86166",
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
              "f04a71a3-399f-4d32-9efc-8a005e989991",
            ],
            "columns": Object {
              "2881fedd-54b7-42ba-8c97-5175dec86166": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of event.category",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": "event.category",
              },
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of undefined",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": undefined,
              },
              "f04a71a3-399f-4d32-9efc-8a005e989991": Object {
                "dataType": "number",
                "isBucketed": false,
                "label": "Maximum of kibana.alert.risk_score",
                "operationType": "max",
                "params": Object {
                  "emptyAsNull": true,
                },
                "scale": "ratio",
                "sourceField": "kibana.alert.risk_score",
              },
            },
            "incompleteColumns": Object {},
            "sampling": 1,
          },
        },
      },
      "textBased": Object {
        "layers": Object {},
      },
    },
    "filters": Array [
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "index": ".alerts-security.alerts-id",
          "key": "kibana.alert.building_block_type",
          "negate": true,
          "type": "exists",
        },
        "query": Object {
          "exists": Object {
            "field": "kibana.alert.building_block_type",
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "host.id",
          "negate": false,
          "params": Object {
            "query": "123",
          },
          "type": "phrase",
        },
        "query": Object {
          "match_phrase": Object {
            "host.id": "123",
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "_index",
          "negate": false,
          "params": Array [
            "signal-index",
          ],
          "type": "phrases",
        },
        "query": Object {
          "bool": Object {
            "minimum_should_match": 1,
            "should": Array [
              Object {
                "match_phrase": Object {
                  "_index": "signal-index",
                },
              },
            ],
          },
        },
      },
    ],
    "internalReferences": Array [],
    "query": Object {
      "language": "kql",
      "query": "host.name: *",
    },
    "visualization": Object {
      "layers": Array [
        Object {
          "categoryDisplay": "default",
          "layerId": "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
          "layerType": "data",
          "legendDisplay": "show",
          "legendPosition": "left",
          "legendSize": "xlarge",
          "metrics": Array [
            "f04a71a3-399f-4d32-9efc-8a005e989991",
          ],
          "nestedLegend": true,
          "numberDisplay": "value",
          "primaryGroups": Array [
            "2881fedd-54b7-42ba-8c97-5175dec86166",
            "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
          ],
        },
      ],
      "shape": "treemap",
    },
  },
  "title": "Alerts",
  "visualizationType": "lnsPie",
}
`;
